api/src/main/java/com/epam/pipeline/manager/cloud/gcp/extractor/GCPCustomMachineExtractor.java

```diff
@@ -35,32 +35,37 @@
 @Component
 public class GCPCustomMachineExtractor implements GCPObjectExtractor {
 
+    private static final int MEGABYTES_IN_GIGABYTE = 1024;
     private static final String CUSTOM_FAMILY = "custom";
+    private static final double EXTENDED_FACTOR = 6.5;
 
     @Override
     public List<AbstractGCPObject> extract(final GCPRegion region) {
         return CollectionUtils.emptyIfNull(region.getCustomInstanceTypes())
                 .stream()
                 .filter(type -> type.getCpu() > 0 && type.getRam() >= 0)
-                .map(type -> {
-                    if (type.getGpu() > 0 && StringUtils.isNotBlank(type.getGpuType())) {
-                        final String name = gpuCustomGpuMachine(type);
-                        return new GCPMachine(name, CUSTOM_FAMILY, type.getCpu(), type.getRam(), type.getGpu(),
-                                type.getGpuType());
-                    } else {
-                        final String name = customCpuMachine(type);
-                        return new GCPMachine(name, CUSTOM_FAMILY, type.getCpu(), type.getRam(), 0, null);
-                    }
-                })
+                .map(this::toGcpMachine)
                 .collect(Collectors.toList());
     }
 
+    private GCPMachine toGcpMachine(final GCPCustomInstanceType type) {
+        final String name = type.getGpu() > 0 && StringUtils.isNotBlank(type.getGpuType())
+                ? gpuCustomGpuMachine(type)
+                : customCpuMachine(type);
+        final double extendedMemory = type.getRam() / type.getCpu() > EXTENDED_FACTOR
```

Comment:
It is a bad practice to compare doubles in such a way. It is better to use, for example, `Precision` class form `apache commons` to compare doubles with some precision.

Reply:
Factors comparison with the `0.1` precision was added.